File: non_abundant_sums/java/non_abundant_sums.java
import java.util.*;

public class non_abundant_sums {
	static ArrayList<Integer> abundant = new ArrayList<Integer>();

	public static void main(String[] args) {
		for (int i=12; i<=28123; i++) {
			if (isAbundant(i)) {
				abundant.add(i);
			}
		}

		int total=0;

		for (int i=1; i<=28123; i++) {
			if (!sumOf2Abundant(i)) {
				total += i;
			}
		}

		System.out.println(total);
	}

	public static boolean sumOf2Abundant(int in) {
		for (int i=0; i<abundant.size() && abundant.get(i) < in; i++) {
			if (isAbundant(in-abundant.get(i).intValue())) {
				return true;
			}
		}
		return false;
	}

	public static ArrayList<Integer> getFactors(int in) {
		ArrayList<Integer> factors = new ArrayList<Integer>();
		factors.add(1);
		int limit = (int)Math.ceil(Math.sqrt(in));
		for (int i=2; i<=limit; i++) {
			if (in % i == 0) {
				factors.add(new Integer(i));
				factors.add(new Integer(in/i));
			}
		}
		Set list = new HashSet(factors);
		factors = new ArrayList<Integer>(list);
		return factors;
	}

	public static boolean isAbundant(int in) {
		if (in <= 0) {
			return false;
		}
		int total=0;

		ArrayList<Integer> factors = getFactors(in);

		for (Integer val : factors) {
			total += val.intValue();
		}
		return total > in;
	}
}

File: non_abundant_sums/php/non_abundant_sums.php
<?php

function getFactors($in): array {
	$factors = Array(1);
	$limit = ceil(sqrt($in));
	for ($i=2; $i <= $limit; $i++) { 
		if ($in % $i == 0) {
			array_push($factors, $i);
			array_push($factors, $in/$i);
		}
	}
	$factors = array_unique($factors);
	return $factors;
}

function isAbundant($in) {
	return array_sum(getFactors($in)) > $in;
}

$abundant = Array();

for ($i=12; $i<=28123; $i++) {
	if (isAbundant($i)) {
		$abundant[] = $i;
	}
}

$canBeSummed = array_map(function($n) { return false; }, range(1, 28123) );;

for ($val1=0; $val1 < count($abundant); $val1++) { 
	for ($val2=$val1; $val2 < count($abundant) && $abundant[$val1]+$abundant[$val2] <= 28123; $val2++) {
		$canBeSummed[$abundant[$val1]+$abundant[$val2]] = true;
	}
}

$total=0;

for ($i=1; $i<=28123; $i++) {
	if ($canBeSummed[$i] != true) {
		$total += $i;
	}
}

var_dump($total);
